### ISSUE TITLE
IndexError when using Beam Search in Chat Completions

### ISSUE THREAD

**Issue by jamestwhedbee** [vllm-project/vllm#2653]
I know this may not be supported given the added complexity, but I just wanted some clarification because I was surprised to see that I could pass these parameters to the server. Right now I non-deterministically hit an IndexError [here](https://github.com/vllm-project/vllm/blob/2e0b6e775756345aa1d39f772c186e00f8c29e92/vllm/entrypoints/openai/api_server.py#L336)

Here is a simple client/server example. I am using v0.2.7

```
python -m vllm.entrypoints.openai.api_server --model mistralai/Mistral-7B-Instruct-v0.1
```

```
from openai import OpenAI

openai_api_key = "EMPTY"
openai_api_base = "http://localhost:8000/v1"

client = OpenAI(
    api_key=openai_api_key,
    base_url=openai_api_base,
)
chat_completion = client.chat.completions.create(
    messages=[{"role": "user", "content": "Hello, world!"}],
    model="mistralai/Mistral-7B-Instruct-v0.1",
    max_tokens=128,
    temperature=0,
    stream=True,
    extra_body={
      "use_beam_search": True,
      "best_of": 3
    }
)
for chunk in chat_completion:
    if chunk.choices[0].delta.content:
        print(chunk.choices[0].delta.content, end="", flush=True)
```

**Comment by simon-mo** [comment]
Ah I just fixed this for completion but not for chat. This is a bug. Thank you for raising this!

**Comment by AnandInguva** [comment]
if this is not solved, I will contribute to this. please let me know.

**Comment by erkintelnyx** [comment]
Hi @simon-mo , is https://github.com/vllm-project/vllm/pull/2529 the mentioned PR?

**Comment by erkintelnyx** [comment]
Completions API accepts batch of prompts while Chat Completions API does not, which confused me a bit about the corresponding proposed solution. For Completions API streaming parameter ends up being False, independent from the stream parameter in the request, [if beam search is used](https://github.com/vllm-project/vllm/blob/main/vllm/entrypoints/openai/serving_completion.py#L317). Would the expected solution be a similar one?

**Comment by simon-mo** [comment]
Yeah I thin the reason is when performing beam search, we can't stream token by token. So disabling is the easiest fix there. This *might* be a legacy constraint given that the generate call can still output incremental tokens results when beam search is used.

**Comment by fuxuelinwudi** [comment]
is there any solutions?

**Comment by senanz** [comment]
Hi, 
Would like to fix the issue.
can someone assign it to me?
